### ISSUE TITLE
Consulting for suggestion

### ISSUE THREAD

**Issue by junfanlin** [thunlp/OpenPrompt#96]
Hi,
Thank you for your quite convenient prompt learning framework. I am new to NLP, especially for prompt learning. I have heard about the success of large-sized MLM. And the most recent advance in zero-shot with prompt learning is so attractive. To experience the "magic" of prompt learning, I have tried the demo of OpenPrompt and self-design a task, i.e.,  selecting the most suitable item from a list according to the description. However, I found that the performance of the self-designed task is very disappointing. I don't know whether I have made some mistakes or it's normal that prompt learning with a manual template performs poorly. Would you mind helping me check the simple code below? Thanks in advance.
```python
# Obtain a PLM

import torch
from openprompt.plms import load_plm
plm, tokenizer, model_config, WrapperClass = load_plm("bert", "bert-large-cased")

# Define a task

from openprompt.data_utils import InputExample
label_words = [
    ["lighter"],
    ["ice"],
    ["water"],
]

dataset = [
    InputExample(
        guid = 0,
        text_a = "What should be used to put out the fire ?"   # expected answer "water"
    ),
    InputExample(
        guid = 1,
        text_a = "What should be used to build a fire ?"  # expected answer "lighter"
    ),
    InputExample(
        guid = 2,
        text_a = "What should be used to strike a spark ?"  # expected answer "lighter"
    ),
    InputExample(
        guid = 3,
        text_b = "This is an intelligence competition .",
        text_a = "What should be used to strike a spark ?"  # expected answer "lighter"
    ),
]

# Define a Template

from openprompt.prompts import ManualTemplate
promptTemplate = ManualTemplate(
    text = '{"placeholder":"text_b"} Question : {"placeholder":"text_a"} Answer : {"mask"} .',
    tokenizer = tokenizer,
)

# Define a Verbalizer

from openprompt.prompts import ManualVerbalizer
promptVerbalizer = ManualVerbalizer(
    num_classes=len(label_words),
    label_words=label_words,
    tokenizer = tokenizer,
)

# Construct a PromptModel

from openprompt import PromptForClassification
promptModel = PromptForClassification(
    template = promptTemplate,
    plm = plm,
    verbalizer = promptVerbalizer,
)
promptModel = promptModel.cuda()

# Define a DataLoader

from openprompt import PromptDataLoader
data_loader = PromptDataLoader(
    dataset = dataset,
    tokenizer = tokenizer,
    template = promptTemplate,
    tokenizer_wrapper_class=WrapperClass,
)

# making zero-shot inference using pretrained MLM with prompt

promptModel.eval()
with torch.no_grad():
    for batch in data_loader:
        batch.cuda()
        logits = promptModel(batch)
        preds = torch.argmax(logits, dim = -1)
        print(label_words[preds])  # results are ["water", "water", "water", "water"], expected results are ["water", "lighter", "lighter", "lighter"]
```

**Comment by ningding97** [comment]
@junfanlin Thanks, hopefully, this issue could help beginners better understand prompt-learning. Prompt-learning essentially uses additional contexts to predict a distribution over the vocabulary, and we only choose those we need as our classification answers. Note that it is still a machine learning problem and *training* is needed in most cases. Although prompt-learning is not black magic that could do all the things, we call still optimize your problem.

Specifically, in your case, you would like to conduct a *zero-shot inference* for your task.  In this setting, the model will tend to predict words that have a high frequency. It is quite obvious that the word "water" is more common than the other words. So we need to *calibrate* the model, and this feature is [implemented](https://github.com/thunlp/OpenPrompt/blob/4a07b071ec137a6ce4a3f80159fe0b6abda87cf1/openprompt/utils/calibrate.py) in OpenPrompt (a [tutorial](https://github.com/thunlp/OpenPrompt/blob/main/tutorial/1.3_calibration.py)). 

Now I simply add the calibration in your code 
```python
cc_logits = calibrate(promptModel, data_loader)
print("the calibration logits is", cc_logits)
promptModel.verbalizer.register_calibrate_logits(cc_logits)
```
The predictions are 
```
['water'], ['lighter'], ['ice'], ['water']
```
Now we have 50% accuracy, not bad, huh? Then I think the word "lighter" may be ambiguous, so I change the label word from "lighter" to "a lighter", the predictions are
```
['water'], ['water'], ['a lighter'], ['a lighter']
```
75%, cool!  But I cannot reach 100% since "water" and "fire" always tend to occur together. As we use a classification manner to solve the task, we should also consider if the task is reasonable. I think the three classes are not representative and the problem may be not clear to the model. After all, *training* is the best way (even few-shot) to alleviate the inherent bias of your language model. 

Enjoy your journey!

**Comment by junfanlin** [comment]
Thank you so much for you quick reply. I have tried your example by myself and it really works! I am going to look into the calibration. Thank you again for your help.
